### PR TITLE
Define :*:.unapply so it mirrors the :*: operator

### PR DESCRIPTION
**Pull request by dwijnand**

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -159,9 +159,9 @@ import BasicJsonProtocol._
 scala> case class Person(name: String, value: Int)
 defined class Person
 
-scala> implicit val personIso = LList.iso(
-         { p: Person => ("name", p.name) :*: ("value", p.value) :*: LNil },
-         { in: String :*: Int :*: LNil => Person(in.head, in.tail.head) })
+scala> implicit val personIso = LList.iso2(
```
Comment (eed3si9n) [inline README.markdown]: This is an aside, but maybe iso2 wasn't such a good name.. Would it compile if it's just called `iso`? If not `isoCurried`?

**Comment by dwijnand** [comment]
Sorry, just realised I was inadvertently relying on args adaptation. Fixed.

File: README.markdown (hunk 1 above)
Comment (dwijnand) [inline README.markdown]: Oh wow, yeah it does compile! Pushing..

**Comment by eed3si9n** [comment]
Synchronizität?

gitter

**Pull request merged**